community/blog/2022-06 .NET 6 finally comes to Arcus Templates in v0.8 release/README.md

```diff
@@ -0,0 +1,112 @@
+
+Azure Functions differ from the usual message pump setup because the message is already being received by the trigger function; whether it is a message on an Azure Service Bus queue or a topic subscription. Since the trigger function already takes over the work of the message pump, we only need to route messages.
+
+Easy extensions on the Azure Functions types allow you to register your message handlers the same way as with a message pump setup. 
+
+```csharp
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+[assembly: FunctionsStartup(typeof(Startup))]
+
+public class Startup : FunctionsStartup
+{
+    public override void Configure(IFunctionsHostBuilder builder)
+    {
+        builder.ConfigureSecretStore(...);
+
+        builder.AddServiceBusMessageRouting()
+               .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>();
+    }
+}
+```
+
+In Azure Service Bus scenarios, a descriptive `IAzureServiceBusMessageRouter` implementation can be injected into your function which gives you access to the message routing and by extension your registered message handlers.
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Abstractions.ServiceBus;
+using Arcus.Messaging.Abstractions.ServiceBus.MessageHandling;
+using Azure.Messaging.ServiceBus;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+public class OrderFunction
+{
+    private readonly IAzureServiceBusMessageRouter _messageRouter;
+    private readonly string _jobId;
+
+    public OrderFunction(IAzureServiceBusMessageRouter messageRouter)
+    {
+        _messageRouter = messageRouter;
+        _jobId = Guid.NewGuid().ToString();
+    }
+
+    [FunctionName("order-processing")]
+    public async Task Run(
+        [ServiceBusTrigger("orders", Connection = "ServiceBusConnectionString")] ServiceBusReceivedMessage message,
+        ILogger log,
+        CancellationToken cancellationToken)
+    {
+        log.LogInformation($"C# ServiceBus queue trigger function processed message: {message.MessageId}");
+
+        AzureServiceBusMessageContext context = message.GetMessageContext(_jobId);
+        MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
+
+        await _messageRouter.RouteMessageAsync(message, context, correlationInfo, cancellationToken);
+    }
```

Comment:
So, this is how we should use the Arcus.Messaging functionality in an Azure Function.
Do we have this information in our documentation as well ?

Reply:
The messaging documentation has this explained on a separated page: https://messaging.arcus-azure.net/Features/message-pumps/service-bus-azure-functions

Reply:
Resolving for now.